youtube_downloader/downloader.py

```
@@ -102,8 +194,39 @@ def download(self, output_file='video.mp4', itag=None, quality=None):
-        response = self.session.get(selected['url'], headers=headers, stream=True)
-        response.raise_for_status()
+        # Handle rate limiting during download
+        retries = 3
+        for attempt in range(retries):
+            try:
+                response = self.session.get(selected['url'], headers=headers, stream=True, timeout=60)
+                
+                # Check for rate limiting
+                if response.status_code == 429:
+                    if self.proxy_manager and attempt < retries - 1:
+                        print(f"\n⚠ Rate limited during download. Rotating proxy...")
```

Ruff (0.14.1)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)